subsys/net/lib/sockets/sockets.c

```diff
@@ -470,10 +600,58 @@ ssize_t zsock_recvfrom(int sock, void *buf, size_t max_len, int flags,
 	return 0;
 }
 
+#ifdef CONFIG_USERSPACE
+Z_SYSCALL_HANDLER(zsock_recvfrom, sock, buf, max_len, flags, src_addr,
+		  addrlen_param)
+{
+	socklen_t addrlen_copy;
+	socklen_t *addrlen_ptr = (socklen_t *)addrlen_param;
+	ssize_t ret;
+	struct net_context *ctx;
+	enum net_sock_type sock_type;
+
+	if (Z_SYSCALL_OBJ(sock, K_OBJ_NET_CONTEXT)) {
+		errno = EBADF;
+		return -1;
+	}
+
+	ctx = INT_TO_POINTER(sock);
+	sock_type = net_context_get_type(ctx);
+
+	if (sock_type != SOCK_DGRAM && sock_type != SOCK_STREAM) {
```

Comment:
Why suddenly validate this for recvfrom()? (Not validated for anything else, e.g. sendto).

Reply:
because the recvfrom implementation doesn't validate properly, it only has an __ASSERT